math/libfaiss/Portfile

```diff
@@ -24,39 +22,134 @@ long_description    Library for efficient similarity search and clustering of \
+                    FAISSSHAREDFLAGS=-dynamiclib \
+                    BLASCFLAGS=-DFINTEGER=int \
+                    "BLASLDFLAGS=-framework Accelerate" \
+                    CPPFLAGS=-I${prefix}/include/libomp
 
-pre-build {
-    touch ${worksrcpath}/makefile.inc
-}
-build.env-append    CC=${configure.cxx}
-build.env-append    "CFLAGS=-fPIC -m64 -Wall -g -O3 -msse4 -mpopcnt -fopenmp \
-                            -Wno-sign-compare -std=c++11 -I/usr/include/malloc/"
-build.env-append    "LDFLAGS=-g -fPIC -fopenmp"
-build.env-append    SHAREDEXT=dylib
-build.env-append    "SHAREDFLAGS=-Wl,-F. -bundle -undefined dynamic_lookup"
-build.env-append    FAISSSHAREDFLAGS=-dynamiclib
-build.env-append    BLASCFLAGS=-DFINTEGER=int
-build.env-append    "BLASLDFLAGS=-framework Accelerate"
 build.target        libfaiss.a libfaiss.dylib
 
-destroot.target
-destroot.destdir
-destroot.cmd        mkdir -p ${destroot}${prefix}/lib && \
-                    cp libfaiss.a libfaiss.dylib ${destroot}${prefix}/lib && \
-                    mkdir -p ${destroot}${prefix}/include/faiss && \
-                    cp -R ./*.h ${destroot}${prefix}/include/faiss/
-
-variant openblas description {Use OpenBLAS instead of Apple's Accelerate framework} {
-    depends_lib-append  port:OpenBLAS
-    build.env-delete    "BLASLDFLAGS=-framework Accelerate"
-    build.env-append    BLASLDFLAGS=${prefix}/lib/libopenblas.dylib
+if {${name} eq ${subport}} {
+    variant openblas description {Use OpenBLAS instead of Apple's Accelerate framework} {
+        depends_lib-append  port:OpenBLAS
+        build.env-delete    "BLASLDFLAGS=-framework Accelerate"
+        build.env-append    BLASLDFLAGS=${prefix}/lib/libopenblas.dylib
+    }
+
+    default_variants +openblas
+}
+
+# create Python subports
+set python_versions {37 38}
+foreach v ${python_versions} {
+    subport py${v}-${github.project} {
+        PortGroup           python 1.0
+        python.default_version  ${v}
+    }
 }
 
-default_variants +openblas
+# Python bindings for supported Python versions
+if {[string match "py*" ${subport}]} {
+    github.setup        ${github.author} ${github.project} ${github.version} ${github.tag_prefix}
+
+    categories-append   math
+    description         Python ${python.branch} bindings for ${description}
+    long_description    This package provides Python ${python.branch} bindings, \
+                        modules, and scripts.
+
+    depends_build-append \
+                        port:py${python.version}-setuptools \
+                        port:swig-python
+    depends_lib-append  port:${name} \
+                        port:py${python.version}-numpy
+
+    post-extract {
+        # fix destroot commands like `$(PYTHON) setup.py install`
+        reinplace -E \
+            "s|^(\[\[:space:]]+\\\$\\(PYTHON\\)\[\[:space:]]+setup\.py\[\[:space:]]+)(install)|\\1 --no-user-cfg \\2 ${destroot.destdir}|" \
+            ${worksrcpath}/python/Makefile
+    }
```

Comment:
I'd prefer this as a patch instead of an (unreadable) `reinplace`

Reply:
Patches are a PITA to create and maintain and this works.

Reply:
you know also that it's the preferred way of doing this in MacPorts.... besides that you should patch that file anyway to make sure it includes the library that was build already instead doing this again (see my comment above as well)

Reply:
See above. A simple symlink works.

Reply:
I looked into a patch but that's a very complicated solution here because the patch involves `${prefix}`. I simple `reinplace` is much easier and transparent.
```
        $(PYTHON) setup.py  --no-user-cfg install --prefix=/opt/local/Library/Frameworks/Python.framework/Versions/3.7 --root=/opt/local/var/macports/build/_opt_local_ports_python_py-faiss/py37-faiss/work/destroot
```

Reply:
The usual thing we do to get around the prefix issue is to patch in a placeholder like @@PREFIX@@ and reinplace the placeholder post-patch. At least we can read that.

Few can tell what this is doing (I don't have the slightest idea what it's up to), it's difficult to debug it when it breaks (if you know it's broken), and the last guy who tried this made some serious errors doing it that were not clear until I ran the thing and generated a diff to see the wreckage.

Reply:
*sigh* no appreciation for BSD sed-specific regex syntax wrapped in tcl…

Changed it to a patch, thanks.